README.md

```diff
@@ -0,0 +1,55 @@
+# What is Weechat-Matrix?
+
+[Weechat](https://weechat.org/) is an extensible chat client.
+
+[Matrix](https://matrix.org/blog/home) is an open network for secure, decentralized communication.
+
+[Weechat-Matrix](https://github.com/poljar/weechat-matrix/) is a Python plugin for Weechat that lets Weechat communicate over the Matrix protocol.
+
+# Project Status
+
+Weechat-Matrix already supports large parts of the Matrix protocol.  See our [TODO](https://github.com/poljar/weechat-matrix/blob/master/TODO.md) list for more details.  Encryption support is still in the works.
+
+# Installation
+
+Installation is easy.  As your regular user, just run: `make install` in this repository directory.
+
+The following python modules must also be available on your system:
+
+- webcolors
+- http\_parser
+
+Note that weechat only supports Python2 OR Python3, and that setting is determined at the time that Weechat is compiled.  Weechat-matrix works with Python2 and Python3, but you will have to install the python dependencies accordingly.  Python2 is a good first guess.
+
+When encryption is ready, the plugin will also depend on the [OLM encryption library](https://git.matrix.org/git/olm/) being properly installed on the system.
+
+# Configuration
+
+Configuration is completed primarily through the Weechat interface.  First start Weechat, and then issue the following commands:
+
+1. Start by loading the weechat-matrix plugin:
+
+    /script load matrix.py test
+
+1. Next set your matrix server:
+
+    /matrix server add myserver matrix.org:8448
```

Comment:
A matrix.org server is already added by default, furthermore, 8448 shouldn't be used (at least with matrix.org not) since it's the federation port, 443 is used with matrix.org.

Reply:
FYI: this is because clients expect to connect to port 443 by default, but the default can be changed by creating an appropriate DNS SRV record.

Reply:
Oh I didn't realize that.  I think I misunderstood the docs or something, because I ran through a matrix tutorial and that was what I understood.